build.sbt

```diff
@@ -1,67 +1,93 @@
 import sbt.Keys.scalaVersion
 import sbt._
+import sbtcrossproject.{CrossType, crossProject}
 
 val monocleVersion = "1.4.0"
 
-lazy val xenomorph = project.in(file("."))
-  .dependsOn(core, argonaut, scalacheck, scodec)
-  .aggregate(core, argonaut, scalacheck, scodec)
+lazy val xenomorph = project
+  .in(file("."))
+  .aggregate(
+    coreJVM, coreJS,
+    scalacheckJVM, scalacheckJS,
+    argonautJVM, argonautJS,
+    scodecJVM
+  )
   .settings(commonSettings)
   .settings(
-    name := "xenomorph"
+    name := "xenomorph",
+    publish := {},
+    publishLocal := {}
   )
 
-lazy val core = project.in(file("modules/core"))
+lazy val core = crossProject(JVMPlatform, JSPlatform)
+  .crossType(CrossType.Pure)
+  .in(file("modules/core"))
   .settings(commonSettings)
   .settings(
     name := "xenomorph-core",
     libraryDependencies ++= Seq(
-      "org.scalaz"                    %% "scalaz-core"     % "7.2.14",
-      "com.chuusai"                   %% "shapeless"       % "2.3.2",
-      "com.github.julien-truffaut"    %% "monocle-core"    % monocleVersion,
+      "org.scalaz"                    %%% "scalaz-core"     % "7.2.18",
+      "com.chuusai"                   %%% "shapeless"       % "2.3.2",
+      "com.github.julien-truffaut"    %%% "monocle-core"    % monocleVersion,
       // test dependencies
-      "com.github.julien-truffaut"  %%  "monocle-generic"  % monocleVersion % "test",
-      "com.github.julien-truffaut"  %%  "monocle-macro"    % monocleVersion % "test",
-      "com.github.julien-truffaut"  %%  "monocle-state"    % monocleVersion % "test",
-      "com.github.julien-truffaut"  %%  "monocle-refined"  % monocleVersion % "test",
-      "com.github.julien-truffaut"  %%  "monocle-law"      % monocleVersion % "test",
-      "joda-time"                   % "joda-time"          % "2.9.4"        % "test",
-      "org.specs2"                  %% "specs2-core"       % "3.9.4"        % "test",
-      "org.specs2"                  %% "specs2-scalacheck" % "3.9.4"        % "test"
+      "com.github.julien-truffaut"  %%%  "monocle-generic"  % monocleVersion % "test",
+      "com.github.julien-truffaut"  %%%  "monocle-macro"    % monocleVersion % "test",
+      "com.github.julien-truffaut"  %%%  "monocle-state"    % monocleVersion % "test",
+      "com.github.julien-truffaut"  %%%  "monocle-refined"  % monocleVersion % "test",
+      "com.github.julien-truffaut"  %%%  "monocle-law"      % monocleVersion % "test",
+      "org.scalatest"               %%% "scalatest"         % "3.0.4"        % "test",
```

Comment:
I had some scary stack traces when using Specs2 in the js. Some JVM method it relies on internally does not seem to have a js equivalent. I preferred switching rather than spending too much time on trying to find a workaround.

Reply:
Okay. I do prefer specs2, but I respect the effort to get this cross-compiling enough that I won't stand in the way of this change. If I can find the time I'll see if I can figure out what's wrong with using specs2 in the .js environment.

Reply:
Good to know, I prefer specs2 too 👍